sky/cli.py

```diff
@@ -133,7 +133,8 @@ def _get_cluster_records_and_set_ssh_config(
     # Update the SSH config for all clusters
     for record in cluster_records:
         handle = record['handle']
-        if handle is not None and handle.cached_external_ips is not None:
+        if (handle is not None and handle.cached_external_ips is not None and
+                'credentials' in record):
```

Comment:
Should we leave a comment about in what cases would `credentials` not exist in the dict?

Reply:
Good point! Just added a comment there. : )